### ISSUE TITLE
SGD documentatiuon detail on g_{t+1}

### ISSUE THREAD

**Issue by moiseevigor** [pytorch/pytorch#41322]
## 📚 Documentation

In the docs https://pytorch.org/docs/master/optim.html#torch.optim.SGD and and inside the class https://pytorch.org/docs/stable/_modules/torch/optim/sgd.html#SGD I notice 

    v_{t+1} & = \mu * v_{t} + g_{t+1},
    p_{t+1} & = p_{t} - \text{lr} * v_{t+1},

But consulting the "On the importance of initialization and momentum in deep learning" by Sutskever et al the update rule for the `v_{t+1}` should be written in this way

    v_{t+1} & = \mu * v_{t} + g_{t},

that is the gradient `g` is taken at time `t` and not `t+1`. Am I wrong?

**Comment by mariosasko** [comment]
Basically, you use `v` from the previous timestamp and `g` from the current timestamp to calculate a new `v` for the current timestamp. Initially **at the timestamp 0**, `v` is set to a vector of zeros (or some other constant), and what are the gradients at that moment?? So `v` starts "one step behind" in comparison to `g` so the first notation seems fine.

**Comment by moiseevigor** [comment]
It is a matter of notation definition if `g_{t+1}` is `g(p_{t+1})` it is not quite what the article says, because at the current step we're defining `p_{t+1}` and in the case of pytorch manual we obtain recursive dependency. But let's see what authors says.
